src/System.Text.Http/System/Text/Http/Obsolete/HttpHeaderBuffer.cs

```diff
@@ -39,64 +39,62 @@ public static void SetFromRestOfSpanToEmpty(this Span<byte> span, int startingFr
             }
         }
 
-        internal static ReadOnlySpan<byte> SliceTo(this ReadOnlySpan<byte> buffer, char terminator, out int consumedBytes)
```

Comment:
Here I am mostly replacing pattern  
` Span Method(..., out int) `
into 
` int Method(..., out Span) `

The first form easily gets into trouble when a ordinary local is passed to the out variable since that leads to unreturnable result.  (since we must assume the resulting span is possibly referring to the local that we passed in).